Atividades/EmCasa/exercicios.js

```diff
@@ -0,0 +1,161 @@
+        this.cor = cor
+        this.consultas = []
+    }
+
+    consultar() {
+        return this.consultas.unshift(new Date())
+    }
+    
+    brincar(){
+        if(this instanceof Gato){
+            console.log("Use uma cordinha")
+        } else if(this instanceof Cachorro){
+            console.log("Brinque de bola")
+        } else if(this instanceof Hamster){
+            console.log("Brinque com a rodinha")
+        } else if(this instanceof Papagaio) {
+            console.log("Cante com ele")
+            this.falar()
+        } else {
+            console.log(`Não atendemos este animal na clínica`)
+        }
+
+    }
+
+    acariciar(){
+        if(this instanceof Gato){
+            this.social ? "ronron" : "tsss"
+        }
+        else  {
+            return console.log (`Você está acaraciando o animal`)
+        }
+    }
+
+    alimentar(alimentos){
+        if(this instanceof Gato && (alimentos == "sache") || (alimentos == "frango") || (alimentos == "peixe")) { // colocar as comidas como estática nos gatos pq são comuns para todos os gatos?
+            this.social == true
+        }
+        else if(this instanceof Papagaio){
+            return console.log(`este ${this.nome} está ${this.falar()}`)
+        } else {
+            return console.log(`Você aliementou ${this.nome} com ${alimentos}`)
+        }
+    }
+
+   
+}
+
+class Domesticos extends Animal {
+    constructor(nome, idade, cor, castrado){
+        super(nome, idade, cor)
+        this.castrado = castrado
+        this.vacinas = []
+    }
+
+    vacinar(vacina) {
+        this.consultar()
+        this.vacinas.push(vacina) 
+        return this.vacinas
+    }
+
+    castrar() {
+        if(this.castrado == true){
+            return console.log(`${this.nome} já foi castrado`)
+        } else {
+            return this.castrado == true
+        }
+    }
+}
+
+class Gato extends Domesticos {
+    constructor(nome, idade, cor, consultas, castrado, vacinas, externo, social){
+        super(nome, idade, cor, consultas, castrado, vacinas)
+        this.externo = externo
+        this.social = social
+    }
+
+    miar(){
+        return console.log("miau") // não sei porque a função está retornando um "undefined" junto com o "miau"
+    }
+}
+
+class Cachorro extends Domesticos{
+    #ferido
+    
+    constructor(nome, idade, cor, consultas, castrado, vacinas, raca, ferido){
+        super(nome, idade, cor, consultas, castrado, vacinas)
+        this.raca= raca
+        this.#ferido = ferido
+    }
+
+    latir(){
+        console.log("au au")
+    }
+
+     brincar(){
+         if(this.ferido == true){
+            console.log(`${this.nome} está ferido e não consegue brincar`)
+         } else {
+            console.log(`Você brincou com ${this.nome}`)
+         }
+     }
+}
+
+class Hamster extends Animal{
+    constructor(nome, idade, cor, consultas, tipo){
+    super(nome, idade, cor, consultas)
+        this.tipo = tipo
+    }
+
+    consultar(){
```

Comment:
Aqui eu senti falta do this.consultas.unshift(new Date()) lindão que vc escreveu lá em cima. Esse método, que sobrescreve o método consultar escrito lá na classe Animal, vai apenas chamar o método brincar e exibir no console o texto "Agora que você brincou com o Hamster, pode seguir com a consulta", sem fazer o push da data pro array de consultas. 

Uma alternativa seria estender o super do consultar, assim:
`consultar( ) {
        this.brincar()
        console.log("Agora que você brincou com o Hamster, pode seguir com a consulta")
        super.consultar( ); //ou  this.consultas.unshift(new Date()). Nesse caso, estaria sobrescrevendo o método e não estendendo
  }      
`